Mesh_3/include/CGAL/Mesh_facet_criteria_3.h

```diff
@@ -41,20 +79,56 @@ class Mesh_facet_criteria_3
   typedef Mesh_3::Criteria<Tr,Visitor> Criteria;
 
   typedef typename Tr::Facet Facet;
-  typedef typename Tr::Geom_traits::FT FT;
+
 
   typedef Mesh_facet_criteria_3<Tr> Self;
 
 public:
   typedef CGAL::Tag_true Has_manifold_criterion;
 
-  /**
-   * @brief Constructor
-   */
-  template < typename Sizing_field, typename Sizing_field2 >
+  /// \name Creation
+  /// @{
+
+#ifdef DOXYGEN_RUNNING
+ /*!
+    returns an object to serve as criteria for facets.
+
+    \param angle_bound is the lower bound for the angle in degrees of the
+    surface mesh facets.
+    \param radius_bound is a uniform upper bound
+    for the radius of the surface Delaunay balls.
+    \param distance_bound is an upper bound for the center-center distances
+    of the surface mesh facets.
+    \param topology is the set of topological constraints
+    which have to be verified by each surface facet. See
+    section \ref Mesh_3DelaunayRefinement for further details.
+    \param min_radius_bound is a uniform lower bound for the radius of
+    the surface Delaunay balls. Only facets with a radius larger than that
+    bound will be refined.
+
+    @note If one parameter is set to 0, then its corresponding
+    criterion is ignored.
+  */
+  Mesh_facet_criteria_3(const FT& angle_bound,
+                        const FT& radius_bound,
+                        const FT& distance_bound,
+                        const Mesh_facet_topology topology = FACET_VERTICES_ON_SURFACE,
+                        const FT& min_radius_bound = 0.);
+
+#endif
+
+
+  /*!
+    Returns an object to serve as criteria for facets. The types `SizingField` and
+    `DistanceField` must
+    be models of the concept `MeshDomainField_3`. The behavior and semantic of the arguments are the same
+    as above, except that the radius and distance bound parameters are
+    functionals instead of constants.
+  */
+  template < typename Sizing_field, typename DistanceField >
```

Comment:
```suggestion
  template < typename Sizing_field, typename Distance_field >
```

Reply:
The naming conventions are CamelCase for template parameters as they are similar to concepts.

Reply:
So `Sizing_field` should become `SizingField.

Reply:
> The naming conventions are CamelCase for template parameters as they are similar to concepts.

I do not know when that was discussed and decided, but I find that particularly bad. In my opinion, template parameters name should be like types.

Otherwise, how to do spell `The template parameter DistanceField must be a model of the concept DistanceField`? If they are both in the same style, they can have the same name, and that is bad. And with real C++20 concepts, that will be even worst:
```c++
template < SizingField Sizing_field, DistanceField Distance_field>
```

If one enforces the same naming convention for template parameter names and concept names, that will be problematic.

Reply:
We are in good company [boost](https://www.boost.org/doc/libs/1_82_0/libs/graph/doc/adjacency_list.html)   [cppreference](https://en.cppreference.com/w/cpp/iterator/next)

Reply:
That is right. Then we just have to find smart names for the template templates, like in [cppreference `std::next`](https://en.cppreference.com/w/cpp/iterator/next):
> `InputIt` must meet the requirements of [_LegacyInputIterator_](https://en.cppreference.com/w/cpp/named_req/InputIterator)